### DISCUSSION TITLE
SSO with Office365

### DISCUSSION THREAD

**Discussion by keesvanbemmel** [directus/directus#9328] [Archive]
Hi all!

Since upgrading the SSO to OpenID I can't seem to log in with any account anymore. I'm using Microsoft as the provider, but I always get "Invalid user credentials." as a response.

How can I check which user identifier MS returns? That way I can check if the correct identifier is set in the user object.

Cheers!

Kees.

Edit:
Did some debugging, microsoft seems to return "nEVRpZrjyVrXQor8cZcredactedOgCQ6bD1sCwz05MCgg" as the identifier. I have no clue how I can know this up front, let alone add users easily with that identifier.

Would switching to oauth fix this?

**Comment by aidenfoxx** [comment]
You can change the `AUTH_<provider>_IDENTIFIER_KEY="email"` if you want to identify your users by email, for example. I've updated the docs to make this more clear but it need to be merged.

By default we use the users OpenID ID for dynamic-registration purposes, which is what you're seeing.

**Reply by keesvanbemmel** [reply]
Thanks! Like a charm!
